dynamiqs/smesolve/smesolve.py

```diff
@@ -63,20 +79,20 @@ def smesolve(
         don't hesitate to
         [open an issue on GitHub](https://github.com/dynamiqs/dynamiqs/issues/new).
 
-    The measured signals $I_k(t)=\dd y_k(t)/\dt$ verifies:
+    The measured signals $I_k(t)=\dd Y_k(t)/\dt$ are defined by:
     $$
-        \dd y_k(t) =\sqrt{\eta_k} \tr{(L_k(t) + L_k(t)^\dag) \rho(t)} \dt + \dd W_k(t).
+        \dd Y_k(t) = \sqrt{\eta_k} \tr{(L_k(t) + L_k^\dag(t)) \rho(t)} \dt + \dd W_k(t).
     $$
 
     Notes:
         Sometimes the signals are defined with a different but equivalent normalisation
-        $\dd y_k'(t) = \dd y_k(t)/(2\sqrt{\eta_k})$.
+        $\dd Y_k'(t) = \dd Y_k(t)/(2\sqrt{\eta_k})$.
 
-    The signals $I_k(t)$ are singular quantities, the solver returns the averaged signals
-    $J_k(t)$ defined for a time interval $[t_0, t_1)$ by:
+    The signals $I_k(t)$ are singular quantities, the solver returns the averaged
+    signals $J_k(t)$ defined for a time interval $[t_0, t_1[$ by:
     $$
-        J_k([t_0, t_1)) = \frac{1}{t_1-t_0}\int_{t_0}^{t_1} I_k(t) \dt
-        = \frac{1}{t_1-t_0}\int_{t_0}^{t_1} \dd y_k(t).
+        J_k([t_0, t_1[) = \frac{1}{t_1-t_0}\int_{t_0}^{t_1} I_k(t)\ \dt
+        = \frac{1}{t_1-t_0}\int_{t_0}^{t_1} \dd Y_k(t).
```

Comment:
I probably missed it, but I thought we were actually returning the `dYt` instead, with the way you wrote it (with the Pytree state). Anyway, just want to make sure :)

Reply:
We indeed accumulate the dYt. The diffrax state at time $t$ is basically

$$\left(\sum_{k=0}^n \mathrm{d}\rho_{k\mathrm{d}t}, \sum_{k=0}^n \mathrm{d}Y_{k\mathrm{d}t}\right)$$

I can add a comment to explain this better too.

But then we do (L104-106 of smediffrax.py):
```python
        # Diffrax integrates the state from t0 to t1. In this case, the state is
        # (rho, dYt). So we recover the signal by simply diffing the resulting array.
        Isave = jnp.diff(integrated_dYt, axis=0)
```
to recover the integrated signal.

Reply:
Actually I think the notations are not very rigorous in the code, I will fix this. Can you read this Notion page and tell me if it's clearer? > https://pierreguilmin.notion.site/SME-diffrax-176db6e9e8cc4bdfba4fb30121642625?pvs=4.

Reply:
The notations are good, but I'm still bothered by something. We simulate the continuous SME for which there is an infinitesimal backaction term because we assume a continuous measurement of the quantum system. But we return the integrated signal, which would correspond to a different SME, with a backaction term that only accounts for binned information about the signal.

Am I missing something?

In other words, I feel like there are two equations:

1. the continuous time SME with infinitesimal backaction and continuous signal,
2. the "binned" SME with binned backaction and discrete signals.

I'm not sure about this second equation: does it even exist/make sense?

Reply:
Let me try to clarify. The discrete-time formulation converges (in law) to the continuous SME as the chosen time step $\mathrm{d}t\to0$. This means that you will get the same statistics for the state and signal as the continuous SME if you take a sufficiently small $\mathrm{dt}$. We _could_ return $\mathrm{d}Y_t/\mathrm{d}t$ for each time step $\mathrm{d}t$ we simulate, but (i) this can easily become super costly in memory if there are lots of timestep, (ii) this quantity should be understood with care for adaptive time-stepping and (iii) it can anyway be reconstructed _if needed_ from the Virtual Brownian Tree holding the sampled Wiener process.

I think having the integrated signal in between user-defined times allows for an easier interpretation and rigorous work. But we could have an option to return the full signal too, and all the time steps, that'd be great actually!